File: src/ResumeTailor.Application/Extraction/Models/FieldExtractionDefinitionRequest.cs


using ResumeTailor.Domain.Extraction;

namespace ResumeTailor.Application.Extraction.Models;

public sealed record FieldExtractionDefinitionRequest(
    int SiteExtractionDefinitionId,
    JobFieldName FieldName,
    string DisplayLabel,
    ExtractionValueType ExtractionType,
    string? AttributeName,
    bool IsRequired,
    int SortOrder
    );

File: src/ResumeTailor.Extension/src/extraction/types/FieldExtractionDefinitionRequest.ts
import type { ExtractionType } from "./ExtractionType";
import type { JobFieldName } from "./JobFieldName";

export type FieldExtractionDefinitionRequest = {
  siteExtractionDefinitionId: number;
  fieldName: JobFieldName;
  displayLabel: string;
  extractionType: ExtractionType;
  attributeName: string | null;
  isRequired: boolean;
  sortOrder: number;
};
